eng/native/functions.cmake

```diff
@@ -1,3 +1,14 @@
+include(CheckCSourceCompiles)
+
+# License: https://github.com/clibs/cmocka/blob/master/cmake/Modules/COPYING-CMAKE-SCRIPTS
+# https://github.com/clibs/cmocka/blob/master/cmake/Modules/CheckCCompilerFlagSSP.cmake
+function(check_c_compiler_flag_ssp _FLAG _RESULT)
```

Comment:
The issue was that `check_c_compiler_flag` would succeed but the linking phase would fail.
The reason is that my toolchain was built without `libssp`, but the flag is valid and recognized by the C compiler.

This check is better because it checks if the program actually links correctly

Reply:
Ah, so without the code with `char buffer[256]; return buffer[argc]=0`, the compiler doesn't inject any checks, thus the linking phase in check_c_source_compiles doesn't see a missing symbol. Makes sense. Thank you for the explanation.